R/function-filtering.R

```diff
@@ -0,0 +1,147 @@
+#'
+#' - `rsd` and `rowRsd` are convenience functions to calculate the relative
+#'  standard deviation (i.e. coefficient of variation) of a numerical vector
+#'  or for rows of a numerical matrix, respectively.
+#'
+#' - `rowDratio` computes the D-ratio or "dispersion ratio," defined as the
+#'  standard deviation for QC (Quality Control) samples divided by the
+#'  standard deviation for biological test samples, for each feature (row) in
+#'  the matrix.
+#'
+#' - `percentMissing` and `rowPercentMissing` determine the percentage of
+#'  missing values in a vector or for each row of a matrix, respectively.
+#'
+#' - `rowBlank` identifies rows (i.e features) where the mean of test samples
+#'  is greater than twice the mean of blank samples. This can highlights
+#'  features that results from contamination in the solvent of the samples.
+#'   Return a `logical` vector
+#'
+#' These functions are based on standard filtering methods described in the
+#' literature, and they are implemented to assist in preprocessing metabolomics
+#' data.
+#'
+#' @param x `numeric` For `rsd`, a numeric vector;
+#'  for `rowRsd`, `rowDratio`, `percentMissing` and `rowBlank`, a numeric
+#'  matrix representing the biological samples.
+#'
+#' @param y `numeric` For `rowDratio` and `rowBlank`, a numeric matrix
+#'  representing the QC samples and blank samples, respectively.
+#'
+#' @param na.rm `logical(1)` indicate whether missing values (`NA`) should be
+#'  removed prior to the calculations.
+#'
+#' @param mad `logical(1)` indicate whether the *Median Absolute Deviation*
+#'  (MAD) should be used instead of the standard deviation. This is suggested
+#'  for non-gaussian distributed data.
+#'
+#' @return  See individual function description above for details.
+#'
+#' @author Philippine Louail, Johannes Rainer
+#'
+#' @md
+#'
+#' @importFrom stats sd mad median
+#'
+#' @name filteringFunctions
+#'
+#' @references
+#'
+#' Broadhurst D, Goodacre R, Reinke SN, Kuligowski J, Wilson ID, Lewis MR,
+#' Dunn WB. Guidelines and considerations for the use of system suitability
+#' and quality control samples in mass spectrometry assays applied in
+#' untargeted clinical metabolomic studies. Metabolomics. 2018;14(6):72.
+#' doi: 10.1007/s11306-018-1367-3. Epub 2018 May 18. PMID: 29805336;
+#' PMCID: PMC5960010.
+#'
+#' @examples
+#'
+#' ## coefficient of variation
+#' a <- c(4.3, 4.5, 3.6, 5.3)
+#' rsd(a)
+#'
+#' A <- rbind(a, a, a)
+#' rowRsd(A)
+#'
+#' ## Dratio
+#' x <- c(4.3, 4.5, 3.6, 5.3)
+#' X <- rbind(a, a, a)
+#' rowDratio(X, X)
+#'
+#' #' ## Percent Missing
+#' b <- c(1, NA, 3, 4, NA)
+#' percentMissing(b)
+#'
+#' B <- matrix(c(1, 2, 3, NA, 5, 6, 7, 8, 9), nrow = 3)
+#' rowPercentMissing(B)
+#'
+#' ## Blank Rows
+#' test_samples <- matrix(c(13, 21, 3, 4, 5, 6), nrow = 2)
+#' blank_samples <- matrix(c(0, 1, 2, 3, 4, 5), nrow = 2)
+#' rowBlank(test_samples, blank_samples)
+#'
+NULL
+
+#' @export
+#' @rdname filteringFunctions
+#'
```

Comment:
please remove the additional `#'` and empty line here.

Reply:
Also below. I find it *cleaner* if the roxygen comments are immediately above the code, e.g.

```r
#' @rdname filterFunctions
rowRsd <- function...
```

instead of 

```r
#' @rdname filterFunctions
#'

rowRsd <- function ...
```